yt/frontends/cholla/data_structures.py

```diff
@@ -0,0 +1,141 @@
+import os
+import weakref
+
+import numpy as np
+
+from yt.data_objects.index_subobjects.grid_patch import AMRGridPatch
+from yt.data_objects.static_output import Dataset
+from yt.funcs import setdefaultattr
+from yt.geometry.grid_geometry_handler import GridIndex
+from yt.utilities.on_demand_imports import _h5py as h5py
+
+from .fields import ChollaFieldInfo
+
+
+class ChollaGrid(AMRGridPatch):
+    _id_offset = 0
+
+    def __init__(self, id, index, level):
+        super().__init__(id, filename=index.index_filename, index=index)
+        self.Parent = None
+        self.Children = []
+        self.Level = level
+
+    def __repr__(self):
+        return "ChollaGrid_%04i (%s)" % (self.id, self.ActiveDimensions)
+
+
+class ChollaHierarchy(GridIndex):
+    grid = ChollaGrid
+
+    def __init__(self, ds, dataset_type="cholla"):
+        self.dataset_type = dataset_type
+        self.dataset = weakref.proxy(ds)
+        # for now, the index file is the dataset!
+        self.index_filename = self.dataset.parameter_filename
+        self.directory = os.path.dirname(self.index_filename)
+        # float type for the simulation edges and must be float64 now
+        self.float_type = np.float64
+        super().__init__(ds, dataset_type)
+
+    def _detect_output_fields(self):
+        h5f = h5py.File(self.index_filename, mode="r")
+        self.field_list = [("cholla", k) for k in h5f.keys()]
+        h5f.close()
+
+    def _count_grids(self):
+        # This needs to set self.num_grids (int)
+        # Probably need to change this.
+        self.num_grids = 1
+
+    def _parse_index(self):
+        self.grid_left_edge[0][:] = self.ds.domain_left_edge[:]
+        self.grid_right_edge[0][:] = self.ds.domain_right_edge[:]
+        self.grid_dimensions[0][:] = self.ds.domain_dimensions[:]
+        self.grid_particle_count[0][0] = 0
+        self.grid_levels[0][0] = 1
+        self.max_level = 1
+
+    def _populate_grid_objects(self):
+        self.grids = np.empty(self.num_grids, dtype="object")
+        for i in range(self.num_grids):
+            g = self.grid(i, self, self.grid_levels.flat[i])
+            g._prepare_grid()
+            g._setup_dx()
+            self.grids[i] = g
+
+
+class ChollaDataset(Dataset):
+    _index_class = ChollaHierarchy
+    _field_info_class = ChollaFieldInfo
+
+    def __init__(
```

Comment:
please check that this method's signature is compatible with existing sibling classes.